syne_tune/blackbox_repository/conversion_scripts/scripts/tabrepo_import.py

```diff
@@ -0,0 +1,270 @@
+        lower=1,
+        upper=5,
+    ),
+    "max_ctr_complexity": uniform(lower=1, upper=5),
+    "one_hot_max_size": choice([2, 3, 5, 10]),
+    "grow_policy": choice(["SymmetricTree", "Depthwise"]),
+}
+
+
+def generate_tabrepo(config_space, bb_name):
+    print(f"generating {bb_name}")
+
+    bb_dict = {}
+
+    # D244_F3_C1530 for full datasets
+    context_name = "D244_F3_C1530_3"
+
+    repo: EvaluationRepository = load_repository(
+        context_name, cache=True, load_predictions=False
+    )
+
+    metrics = repo.metrics(datasets=repo.datasets(), configs=repo.configs())
+    metrics = metrics[
+        metrics.index.get_level_values("framework").str.split("_").str[0]
+        == bb_name.split("_")[1]
+    ]
+    for dataset_name, group in metrics.groupby("dataset"):
+        print(f"Processing dataset: {dataset_name}")
+        hyperparameters_configurations = {}
+        for framework in group.index.to_frame(index=False)["framework"].values:
+            hyperparameters_configurations[framework] = repo.config_hyperparameters(
+                config=framework
+            )
+        bb_dict[dataset_name] = convert_dataset(
+            config_space, group, hyperparameters_configurations
+        )
+
+    with catchtime("saving to disk"):
+        serialize(
+            bb_dict=bb_dict,
+            path=repository_path / bb_name,
+            metadata={metric_elapsed_time: METRIC_ELAPSED_TIME},
+        )
+
+
+def convert_dataset(config_space, evaluations, configurations):
+    # names of hyperparameters
+    hp_cols = list(config_space.keys())
+    # number of hyperparameters
+    n_hps = len(hp_cols)
+    # number of evaluations
+    n_evals = len(evaluations.xs(0, level="fold"))
+    # initialize hyperparameter array with shape(n_evals, n_hps)
+    hps = np.zeros((n_evals, n_hps), dtype=object)
+
+    n_seeds = 3
+
+    for i, config in enumerate(configurations):
+        arr = []
+        for key in config_space.keys():
+            # this handles cases, where not all hps are set
+            if key in configurations[config].keys():
+                # this avoids type errors in fastparquet conversion, applies only to RandomForrest and ExtraTrees
+                if key == "max_features":
+                    arr.append(str(configurations[config][key]))
+                else:
+                    arr.append(configurations[config][key])
+            # set hp to None if not existent
+            else:
+                arr.append(None)
+        hps[i] = arr
+
+    hyperparameters = pd.DataFrame(data=hps, columns=hp_cols)
+    # can I just rename time_train to metric elapsed time or does the data need to be changed somehow?
+    objective_names = [
+        "metric_error",
+        "metric_error_val",
+        "metric_elapsed_time",
+        "metric_time_infer_s",
+        "metric_rank",
+    ]
+    n_objectives = len(objective_names)
+
+    objective_evaluations = []
+
+    for i, fold in enumerate([0, 1, 2]):
```

Comment:
Here a comment would be useful too, something like: "We interpret fold, dataset split, as seed of the blackbox, we build a tensor by..."